articles/quickstart/spa/angular-beta/_includes/_centralized_login.md

```diff
@@ -0,0 +1,459 @@
+
+Finally, to show this component on the screen, open the `src/app/app.component.html` file and replace the default UI with the following:
+
+```html
+<app-navbar></app-navbar>
+
+<router-outlet></router-outlet>
+```
+
+> **Checkpoint**: Run the application now using `npm start`. The login button should be visible, and you should be able to click it to be redirected to Auth0 for login.
+
+## Handle Login Redirects
+
+To log the user into your application, the redirect from Auth0 should be handled so that a proper login state can be achieved. To do this, the `handleLoginRedirect` SDK method should be called when Auth0 redirects back to your application. This can be done by using a separate component, whose sole job is to handle the redirect callback.
+
+Use the Angular CLI to create a new component called `Callback`:
+
+```bash
+ng generate component Callback
+```
+
+Open the newly-generated `src/app/callback/callback.component.ts` file and replace its contents with the following:
+
+```js
+import { Component, OnInit } from '@angular/core';
+import { AuthService } from 'src/app/auth.service';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-callback',
+  templateUrl: './callback.component.html',
+  styleUrls: ['./callback.component.css']
+})
+export class CallbackComponent implements OnInit {
+  constructor(private authService: AuthService, private router: Router) {}
+
+  async ngOnInit() {
+    const client = await this.authService.getAuth0Client();
+
+    // Handle the redirect from Auth0
+    const result = await client.handleRedirectCallback();
+    
+    // Get the URL the user was originally trying to reach
+    const targetRoute =
+      result.appState && result.appState.target ? result.appState.target : '';
+
+    // Update observables
+    this.authService.isAuthenticated.next(await client.isAuthenticated());
+    this.authService.profile.next(await client.getUser())
+
+    // Redirect away
+    this.router.navigate([targetRoute]);
+  }
+}
+```
+
+Once again the `AuthService` class is injected so that the component can gain access to the Auth0 SDK client. The key line is `await client.handleRedirectCallback`, which takes the processes the data in the callback from Auth0 and retrieves the tokens. It also sets some internal state that can be used to determine whether the user is authenticated or not.
+
+Notice also the the component updates the observable properties inside the `AuthService` instance. This means that other components that are watching these observables can update their UI at the right time, when the user logs in. This component also handles the scenario where the user is trying to reach some content that is unavailable to unauthenticated users. In this case, `targetRoute` holds the URL they were trying to reach before redirecting to Auth0, and is used to redirect the user to the right place after the callback.
```

Comment:
> the the

Reply:
> ... at the right time, when the user logs in.

Consider:

> ... at the right time: when the user logs in

Reply:
> This component also handles the scenario where the user is trying to reach some content that is unavailable to unauthenticated users.

Feels awkward ...

Reply:
@joshcanhelp Rewrote the last couple of sentences, hopefully it reads better.